src/streaming-client.ts

```diff
@@ -0,0 +1,156 @@
+import * as fs from 'fs';
+import * as sockets from 'websocket';
+import * as events from 'events';
+import { Duplex, PassThrough } from 'stream';
+
+import { AudioConfig } from './models/streaming/AudioConfig';
+import { BufferedDuplex } from './models/streaming/BufferedDuplex';
+import { 
+    StreamingResponse, 
+    StreamingHypothesis, 
+    StreamingConnected 
+} from './models/streaming/StreamingResponses';
+
+/**
+ * Client which handles a streaming connection to the Rev.ai API.
+ *
+ * @param accessToken {string} Access token associated with the user's account
+ * @param config {AudioConfig} Configuration of the audio the user will send from this client
+ * @param version (optional) {string} Version of the Rev.ai API the user wants to use
+ * 
+ * @event httpResponse emitted when the client fails to start a websocket connection and
+ *      receives an http response. Event contains the http status code of the response.
+ * @event connectFailed emitted when the client fails to begin a websocket connection and
+ *      received a websocket error. Event contains the received error.
+ * @event connect emitted when the client receives a connected message from the API. Contains
+ *      the StreamingConnected returned from the API.
+ * @event close emitted when the connection is properly closed by the server. Contains the
+ *      close code and reason.
+ * @event error emitted when an error occurs in the connection to the server. Contains the
+ *      thrown error.
+ *
+ * @example
+ * const audioConfig = new AudioConfig(<YOUR-CONTENT-TYPE-PARAMS>);
+ * const token = <YOUR-ACCESS-TOKEN>;
+ * var client = new RevAiStreamingClient(token, audioConfig);
+ * client.on('close', (code, reason) => {
+ *     console.log(`Connection closed, ${code}: ${reason}`);
+ * });
+ * client.on('httpResponse', code => {
+ *     console.log(code);
+ * })
+ * client.on('connectFailed', error => {
+ *     console.log(error);
+ * })
+ * client.on('connect', connectionMessage => {
+ *     console.log(connectionMessage);
+ * })
+ * 
+ * var stream = client.start();
+ * 
+ * var file = fs.createReadStream(<YOUR-AUDIO-FILE>);
+ * stream.on('data', data => {
+ *     console.log(data);
+ * });
+ * stream.on('end', function () {
+ *     console.log("End of Stream");
+ * });
+ * 
+ * file.pipe(stream);
```

Comment:
This example might have better visibility in a `Getting Started` section of the readme.

Reply:
I don't plan on writing the readme until we add streaming documentation to the website since it will definitely need to reference that. Since this isn't meant to be very visible until then I think keeeping the example here is good

Reply:
I liked what alex was saying about including example apps in our repository. The examples won't be packaged into the distributions, but they can be viewed/forked on github.